### PR TITLE
[ASP.NET Core] Add additional `http.server.duration` metric attributes

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Towards #3373 

## Changes

As per spec https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#attributes

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 1]
```diff
@@ -2,14 +2,18 @@
 
 ## Unreleased
 
+* Added `net.host.name` and `net.host.port` dimensions to `http.server.duration`
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:5]: Mention something about the dimension `http.host` no longer being populated for the metric.

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 2]
```diff
@@ -2,14 +2,18 @@
...
+metric.
+[#3928](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3928)
+
+* **Breaking change** The `http.server.duration` metric's `http.target`
+attribute is replaced with `http.route` attribute.
+([#3903](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3903))
+
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:12]: These entries could be reorganized to make it more readable. Something like this:
* Users migrating from `1.0.0-rc9.9` to this version will see the following breaking changes:
   -
   -
   -

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md (hunk 1 above)
Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md]: my bad, this is a breaking change. Fixed

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md (hunk 2 above)
Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:14]: Updated

**Pull request merged**